### ISSUE TITLE
The ESM symbol from eslint-utils ReferenceTracker should be typed

### ISSUE THREAD

**Issue by MageJohn** [typescript-eslint/typescript-eslint#3527]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
import { ASTUtils } from "@typescript-eslint/experimental-utils";
const { ReferenceTracker } = ASTUtils;

const ESM = ReferenceTracker.ESM;
```

**Expected Result**

No type errors.

**Actual Result**

A type error, because there is no `ESM` key on the exported `ReferenceTracker` interface from `ast-utils`.

**Additional Info**

This seems to be an undocumented export, but you can see it being defined in eslint-utils [here](https://github.com/mysticatea/eslint-utils/blob/c3b1d38328dc3a5278b208a58186a0735315f58f/src/reference-tracker.js#L440). It's apparently used to mark a traceMap as being an ES module.

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `4.26.1` |
| `TypeScript`                            | `4.3.2` |
| `node`                                  | `12.22.0` |

**Comment by bradzacher** [comment]
We accept all contributions.
Don't be afraid to raise one to add it!
